patricia_trie/src/nibbleslice.rs

```diff
@@ -147,12 +152,29 @@ impl<'a> NibbleSlice<'a> {
  	/// How many of the same nibbles at the beginning do we match with `them`?
 	pub fn common_prefix(&self, them: &Self) -> usize {
 		let s = min(self.len(), them.len());
-		let mut i = 0usize;
-		while i < s {
-			if self.at(i) != them.at(i) { break; }
-			i += 1;
+		if s % 2 == 0 && s > 2 && self.offset % 2 == 0 && them.offset % 2 == 0 {
```

Comment:
I'm not very familiar with this code but can't we do better than this? AFAICT the offsets need to be byte-aligned for us to be able to do bytewise comparison, but if the length doesn't terminate at a byte-boundary we could still do byte-by-byte comparison until the last nibble right?

Reply:
I think you are right and that we can; I just didn't have the time to dig deeper to figure out why the tests kept failing. I think the problem is that the "odd" nibble is shoved into the *first* byte, so for odd-length offsets to work here I think we'd have to pluck out the first nibble from the first byte and use it in some smart way. I considered doing the whole work backwards instead to get around this, but yeah, in the end I just gave up.

I was also hoping for a reviewer smarter than me coming up with a better way! ;)

Reply:
match (self.offset % 2, them.offset % 2) {
    (0, 0) => // byte compare
    (1, 1) => // check first item then byte compare
    _ => // iter compare
}
// ... then do the same with the suffixes if there is no break.